### ISSUE TITLE
Unable to import .ts files in index.ts

### ISSUE THREAD

**Issue by italodirenzo93** [team-innovation/vue-sfc-rollup#80]
In a fresh TypeScript with Vue2 project and using the default `rollup.config.js` , attempting to run the `build` command with an index file that looks like this:

```js
export { default as MyComponent } from './MyComponent.vue';
export * from './options';
```

Results in this error:
```
[!] Error: Could not resolve './options' from src/lib-components/index.ts
```

Both `MyComponent.vue` and `options.ts` reside in the `lib-components` folder. I have poked around at this for hours an cannot understand why my `.ts` file are unable to be resolved the same way as my `.vue` files.

**Comment by mgdodge** [comment]
Do you have a sample repo?

**Comment by italodirenzo93** [comment]
This is in a client's repository but I can make a sanitized copy for reproduction purposes.

**Comment by italodirenzo93** [comment]
@mgdodge https://github.com/italodirenzo93/vue-sfc-rollup-ts-import

**Comment by pareshchouhan** [comment]
I have the same issue, except when I build it. It shows an error. Not while importing. 
During builds I have to specify .ts type and during development I had to remove all of those types.

**Comment by mgdodge** [comment]
When using the vue cli, you can use paths like `import foo from '@/foo';` and it will automatically resolve them to `src/foo`. To maintain compatibility, here we use `@rollup/plugin-alias` to enable the same path aliasing. If you change your import from `./options` to `@/lib-components/options` then the imports seem to resolve just fine. So it seems the use of this alias capability is somehow preventing resolution of non-aliased paths.

You can add the following to the rollup config to resolve non-aliased paths, right after the `alias(...)` plugin (about line 38 in your demo repo `rollup.config.js` file):

```
      resolve({
        extensions: ['.js', '.jsx', '.ts', '.tsx', '.vue'],
      }),
```

Thanks for making me aware of this bug, I'll get it patched shortly.

---

I noticed one other unrelated thing which might be worth noting: your sample here is importing the `options.ts` file as part of the `lib-components/index.ts` barrel file. This might be problematic as the `install` function in `entry.esm.ts` iterates over everything exported by this barrel file and attempts to run it through `Vue.component()`. Obviously, the `options.ts` exports there are not vue components.

I would recommend that lib-components _only_ contain vue components. You would then either export the `options.ts` file itself from the `entry.esm.ts` file directly, or create a separate barrel file for non-component exports, and re-export that barrel in the same way. For example:

```
export * from '@/lib-components/index';

// Re-export options directly
export * from '@/lib-components/options';

// -- OR -- create new barrel file for non-component exports
export * from '@/lib-utils/index';
```

**Comment by mgdodge** [comment]
This is now resolved in v4.0.3

**Comment by mgdodge** [comment]
Testing another bug, and noticing that the way I suggested to resolve this here seems to prevent the default configuration from compiling css. It's a plugin order thing - make sure the resolve comes AFTER the vue plugin itself, not after the alias.

**Comment by pareshchouhan** [comment]
@mgdodge , for using ts classes I had to import   plugins: ['@babel/plugin-proposal-class-properties'] in babel.config.js and 
```
"babel-plugin-transform-class-properties": "^6.24.1",
```

**Comment by mgdodge** [comment]
Released a patch yesterday, v4.0.4, which uses the proper plugin order.
